src/main/scala/org/renci/cam/LookupService.scala

```diff
@@ -0,0 +1,420 @@
+    * @param objectTriples
+    *   All object triples for the identifier being looked up (i.e. triples in the triplestore in the shape `?s ?p normalizedId`)
+    */
+  case class Result(
+    queryId: String,
+    normalizedIds: List[LabeledIRI],
+    biolinkPredicates: Map[String, Set[LabeledIRI]],
+    relations: Seq[Relation],
+    subjectTriples: Seq[ResultTriple],
+    objectTriples: Seq[ResultTriple]
+  )
+
+  /** An error returned by calling `/lookup`. */
+  case class Error(
+    code: String,
+    message: String
+  )
+
+  /* CONTROLLER */
+  type LookupEndpointParams = (String, Int, Option[String], String)
+  type LookupEndpoint = Endpoint[LookupEndpointParams, Error, Result, Any]
+
+  /* Helper methods */
+
+  /** Returns a list of Relations for a particular subject.
+    *
+    * TODO: in a future version, this will call itself recursively so we can fill in objRelations.
+    *
+    * @param qualifiedIds
+    *   The subject IRIs to find relations for.
+    */
+  def getRelations(
+    qualifiedIds: Set[String]): ZIO[ZConfig[AppConfig] with HttpClient with Has[BiolinkData], Throwable, Iterable[Relation]] =
+    for {
+      biolinkData <- biolinkData
+
+      subjectIRIs = {
+        implicit val iriKeyDecoder: KeyDecoder[IRI] = Implicits.iriKeyDecoder(biolinkData.prefixes)
+
+        qualifiedIds.map(qualifiedId => iriKeyDecoder(qualifiedId).get)
+      }
+
+      relationsQuery =
+        sparql"""PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+               PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+
+               SELECT DISTINCT ?subj ?subjLabel ?p ?pLabel ?obj ?objLabel ?g {
+                  ?s <http://www.openrdf.org/schema/sesame#directType> ?subj .
+                  ?subj <http://www.w3.org/2000/01/rdf-schema#subClassOf> ?subjClass .
+                  VALUES ?subjClass { ${subjectIRIs.asValues} } .
+                  
+                  ?o <http://www.openrdf.org/schema/sesame#directType> ?obj .
+                  ?obj <http://www.w3.org/2000/01/rdf-schema#subClassOf> <https://w3id.org/biolink/vocab/NamedThing> .
+
+                  GRAPH ?g {
+                    ?s ?p ?o
+                  }
+
+                  OPTIONAL { ?subjClass rdfs:label ?subjLabel }
+                  OPTIONAL { ?p rdfs:label ?pLabel }
+                  OPTIONAL { ?obj rdfs:label ?objLabel }
+               }"""
+      relationsResults <- SPARQLQueryExecutor.runSelectQuery(relationsQuery.toQuery)
+      preds = relationsResults.map(_.getResource("p").getURI).map(IRI(_))
+      biolinkRelationMap <- QueryService.mapRelationsToLabelAndBiolink(preds.toSet)
+
+      // We want to group these by object, so we don't return a gazillion predicates for each result.
+      objectMap = relationsResults.groupBy(_.getResource("obj").getURI)
+
+      relations = objectMap map { case (obj, results) =>
+        val predResults = results map { res =>
```

Comment:
We can discuss, but I only really use infix syntax for DSL situations.

Reply:
In turns out this wasn't entirely my fault -- this is how scalafmt formats this with our settings. I modified .scalafmt.conf (3249bcfd7b1e0) to add ApplyInfix and re-ran scalafmt, which partially fixed this to `objectMap.map { case (obj, results) =>` (a1826d981b0). Do you think that's good enough, or should we try to figure out how to modify .scalafmt.conf to add a curly bracket there?

Reply:
That looks good!